dns-server/src/lib.rs

```diff
@@ -86,3 +91,79 @@ pub async fn start_servers(
 
     Ok((dns_server, dropshot_server))
 }
+
+/// An in-memory DNS server running on localhost.
+///
+/// Intended to be used for testing only.
+pub struct InMemoryServer {
+    /// Server storage dir
+    pub storage_dir: tempfile::TempDir,
+    /// DNS server
+    pub server: dns_server::ServerHandle,
+    /// Dropshot server
+    pub dropshot_server: dropshot::HttpServer<http_server::Context>,
+}
```

Comment:
This was "rolled-by-hand" in the Sled Agent before this PR.

For test / simulated workloads, it seems pretty reasonable to provide this low-touch implementation.

Reply:
I intended for this to be a really thin wrapper around the production DNS server - the only difference is the `dns_server::storage::Store` argument, which is created from this `tempfile::TempDir`.

If you compare what I'm doing here with tests, I intended for it to be the same. For example, see:

https://github.com/oxidecomputer/omicron/blob/9b85b6190cf58e7860a04471f506fbaecabd219f/nexus/src/app/background/init.rs#L227-L256

- This creates a `TempDir::new()`
- It uses it as a `dns_server::storage::Store`
- It calls `start_servers`, and creates a new dropshot server

`InMemoryServer` is supposed to just be a convenience wrapper around this same thing.

Reply:
Sorry about that!  I didn't look at the implementation closely enough and was confused by the name.  That sounds perfect, but can we not call it "in-memory"?  If you're trying to convey that it goes away, maybe "Transient"?